### PR TITLE
Extended the CFNumber data formatter to handle High Sierra.

### PR DESCRIPTION
**Pull request by scallanan**
<rdar://problem/32780109>

### CODE REVIEW

**Comment by scallanan** [comment]
@swift-ci please test

**Comment by scallanan** [comment]
Whoops, got the Foundation version number wrong.  Thanks @jasonmolenda for catching that.

**Comment by scallanan** [comment]
@swift-ci please test

**Comment by scallanan** [comment]
@swift-ci please test

**Comment by scallanan** [comment]
@swift-ci please test OS X

**Comment by scallanan** [comment]
@swift-ci please test

**Comment by scallanan** [comment]
@swift-ci please test

File: source/Plugins/Language/ObjC/Cocoa.cpp [hunk 1]
```diff
@@ -500,7 +500,7 @@ bool lldb_private::formatters::NSNumberSummaryProvider(
               process_sp->GetObjCLanguageRuntime());
 
       const bool new_format =
-          (runtime && runtime->GetFoundationVersion() >= 1100);
+          (runtime && runtime->GetFoundationVersion() >= 1400);
```
Comment (jrose-apple) [inline source/Plugins/Language/ObjC/Cocoa.cpp]: This doesn't look right. What happens for Foundation versions 1100..<1400?

Comment (jrose-apple) [reply source/Plugins/Language/ObjC/Cocoa.cpp]: Oops, I missed that this was all new code, never mind.

**Comment by dcci** [comment]
Closing as this is inactive and nobody is picking this up. The formatters on high Sierra seem to work, this is a PR against an old branch, etc..

**Pull request closed without merging**